### DISCUSSION TITLE
PeakShaving with GW10K-ET

### DISCUSSION THREAD

**Discussion by C74A** [mletenay/home-assistant-goodwe-inverter#323] [Q&A]
Hi,

I have ellaborated with PeakShaving functionality. I get it working somehow, but I have one strange behavior. When the power needed is higher than the limit I have defined in SEMS portal and the battery start to discharge. The power from the battery equals the total consumed power and not the amount of power above my defined limit. Ex: I set the limit to 3000W. The consumption goes beyond this and is 5000W. In this case the battery discharges with 5000W and not 2000W. I think this can be related to that the basic operation mode. in HA the work mode code is "1" and I think this means that the "operation mode" of the inverter is priority of self-consumption and this overrides the peakshaving settings. (Inverter manual from GoodWee) Is this correct understanding? If so, what "Operation Mode" is needed for peakshaving and how is this modified? What other consequences might this Operation Mode provide for my system?

Firmware: 04029-09-S11 / 02041-22-S00
Hårdvara: ETU 9010K

**Comment by dubbleeagle** [comment]
Your parameters should look like this:
- Start Time = 00:00
- End Time = 23:59
- Repeat = Every day (you have to check them all)
- Power limit = 3 kW
- SOC = 50%
With these parameters, it will use all requested power (5000W) from the battery when the SOC is above 50%. Once the battery level drops to 50%; it will use 3000W from the grid, and 2000W from the battery. When your consumption drops to 1000W (and SOC is < 50%), you will get 3000W from the grid: 1000W for household and 2000W to recharge the battery up to 50%.
